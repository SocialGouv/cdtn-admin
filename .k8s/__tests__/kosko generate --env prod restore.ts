//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --prod restore", async () => {
  process.env.SOURCE_CONTAINER = "source-container";
  process.env.DESTINATION_CONTAINER = "destination-container";
  process.env.BACKUP_DB_OWNER = "some-owner";
  process.env.BACKUP_DB_NAME = "some-database";
  process.env.BACKUP_DB_FILE = "some-backup.sql.gz";
  expect(
    await getEnvManifests("prod", "restore", {
      ...project("cdtn-admin").prod,
    })
  ).toMatchSnapshot();
});
