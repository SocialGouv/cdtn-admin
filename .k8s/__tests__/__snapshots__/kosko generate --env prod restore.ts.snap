// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --prod restore 1`] = `
"metadata:
  name: restore-container-8843083e
  namespace: cdtn-admin-secret
spec:
  backoffLimit: 0
  template:
    metadata: {}
    spec:
      containers:
        - command:
            - sh
            - '-c'
            - >+


              [ ! -z $SOURCE_CONTAINER ] || (echo \\"No SOURCE_CONTAINER\\"; exit 1)

              [ ! -z $SOURCE_ACCOUNT_NAME ] || (echo \\"No SOURCE_ACCOUNT_NAME\\";
              exit 1)

              [ ! -z $SOURCE_ACCOUNT_KEY ] || (echo \\"No SOURCE_ACCOUNT_KEY\\";
              exit 1)

              [ ! -z $DESTINATION_CONTAINER ] || (echo \\"No
              DESTINATION_CONTAINER\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_NAME ] || (echo \\"No
              DESTINATION_ACCOUNT_NAME\\"; exit 1)

              [ ! -z $DESTINATION_ACCOUNT_KEY ] || (echo \\"No
              DESTINATION_ACCOUNT_KEY\\"; exit 1)



              echo \\"starting restore container $SOURCE_CONTAINER into
              $DESTINATION_CONTAINER\\"


              # copy container

              az storage copy -r   --source-account-name
              \\"$SOURCE_ACCOUNT_NAME\\"   --source-account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --source-container \\"$SOURCE_CONTAINER\\"  
              --account-name \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --destination-container
              \\"$DESTINATION_CONTAINER\\"


              # get original permission

              PUBLIC_ACCESS=\`az storage container show-permission  
              --account-name \\"$SOURCE_ACCOUNT_NAME\\"   --account-key
              \\"$SOURCE_ACCOUNT_KEY\\"   --name $SOURCE_CONTAINER | jq -r
              \\".publicAccess\\"\`


              # set original permission

              az storage container set-permission   --name
              $DESTINATION_CONTAINER   --account-name
              \\"$DESTINATION_ACCOUNT_NAME\\"   --account-key
              \\"$DESTINATION_ACCOUNT_KEY\\"   --public-access $PUBLIC_ACCESS

          env:
            - name: SOURCE_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  name: azure-cdtnadminprod-volume
                  key: azurestorageaccountname
            - name: SOURCE_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  name: azure-cdtnadminprod-volume
                  key: azurestorageaccountkey
            - name: DESTINATION_ACCOUNT_NAME
              valueFrom:
                secretKeyRef:
                  name: azure-cdtnadmindev-volume
                  key: azurestorageaccountname
            - name: DESTINATION_ACCOUNT_KEY
              valueFrom:
                secretKeyRef:
                  name: azure-cdtnadmindev-volume
                  key: azurestorageaccountkey
            - name: SOURCE_CONTAINER
              value: source-container
            - name: DESTINATION_CONTAINER
              value: destination-container
          envFrom: []
          image: 'mcr.microsoft.com/azure-cli:2.15.1'
          imagePullPolicy: IfNotPresent
          name: restore-db
          resources:
            limits:
              cpu: 300m
              memory: 512Mi
            requests:
              cpu: 50m
              memory: 64Mi
      restartPolicy: Never
apiVersion: batch/v1
kind: Job
"
`;
